### PR TITLE
Replace yield from with await...

### PR DESCRIPTION
**Pull request by terrycain**
#281 Point 2 Replace yield from with await.

Well this resulted in a nastier diff than I expected :D.

Have replaced the majority of the `yield from` with `await` and `@asyncio.coroutine` with `async`

This is more a placeholder for when you want to purge the yield from. Its not 100% finished but if you fancy reading through it, by all means :)

### CODE REVIEW

File: aiomysql/sa/result.py [hunk 1]
```diff
@@ -340,18 +336,18 @@ def close(self):
 
         if not self._closed:
             self._closed = True
-            yield from self._cursor.close()
+            await self._cursor.close()
             # allow consistent errors
             self._cursor = None
             self._weak = None
 
-    def __iter__(self):
```
Comment (terrycain) [inline aiomysql/sa/result.py]: Yeah this stopped working... not too sure why

**Comment by jettify** [comment]
Can you please split this PR in 2 parts: just port tests, without library change and subsequent PR with library change? Should be easy to do, since you have one commit per file. I can help you with splitting.

Reason for 2 PR,  I just want to be extra caution that move to `async def` does not introduce new issue.

**Comment by jettify** [comment]
Other question, is what minimum version of python3.5 we should support?

**Comment by terrycain** [comment]
Sure makes much more sense. Any recommendations for splitting the PR, as normally I resort to mashing git commands until it looks ok :smile:.

As for Python3.5 versions, I don't know enough about the differences between patch versions, do you know of anything that we should be cautious of? Otherwise, 3.5.2 is on Ubuntu, so assuming that has a fair market share, that's not a bad minimum.

**Comment by jettify** [comment]
I was thinking about 3.5.3, since older versions has annoying bug with with `asyncio.get_event_loop()`, it may return other loop not currently running in some strange cases.

**Comment by terrycain** [comment]
LOL yeah that is an annoying bug. Sure 3.5.3 is fine by me, im on 3.6 :D

**Comment by jettify** [comment]
cherry picked tests locally to make sure all set

**Comment by jettify** [comment]
thanks!

**Pull request merged**